packages/astro/test/csrf-protection.test.js

```diff
@@ -176,6 +176,34 @@ describe('CSRF origin check', () => {
 		});
 	});
 
+	it("return a 200 when the origin doesn't match but calling HEAD", async () => {
+		let request;
+		let response;
+		request = new Request('http://example.com/api/', {
+			headers: { origin: 'http://loreum.com', 'content-type': 'multipart/form-data' },
+			method: 'HEAD',
+		});
```

Comment:
The test is failing because you're hitting an endpoint, which exposes only the `GET` method.

You should:
- create an astro component and hit that route
- expand the `api.ts` file and add the rest of the verbs to test

Reply:
Thanks for the pointers!